[KubeRay][Doc] KubeRay doc migration (5/n) (#39450)
ray-project/ray: doc/source/serve/architecture.md

- When replica actors fail, the Controller actor will replace them with new ones.
- When the HTTP proxy actor fails, the Controller actor will restart it.
- When the Controller actor fails, Ray will restart it.

> - When using the [KubeRay RayService](kuberay-rayservice-quickstart), KubeRay will recover crashed nodes or a crashed cluster.  Cluster crashes can be avoided using the [GCS FT feature](kuberay-gcs-ft).

Reviewer: ```suggestion
- When using the [KubeRay RayService](kuberay-rayservice-quickstart), KubeRay recovers crashed nodes or a crashed cluster.  You can avoid cluster crashes using the [GCS FT feature](kuberay-gcs-ft).
```

> - If not using KubeRay, when the Ray cluster fails, Ray Serve cannot recover.

Reviewer: ```suggestion
- If you're not using KubeRay, when the Ray cluster fails, Ray Serve cannot recover.
```

> When a machine hosting any of the actors crashes, those actors will be automatically restarted on another

Reviewer: ```suggestion
When a machine hosting any of the actors crashes, those actors are automatically restarted on another
```
